File: 0567-permutation-in-string/0567-permutation-in-string.py
class Solution:
    def checkInclusion(self, s1: str, s2: str) -> bool:
        Len = (len(s2) - len(s1)+1)
        str1 = ''.join(sorted(s1))
        start = 0
        while start <= Len:
            newStri = ''.join(sorted(s2[start:start+len(s1)]))
            if str1 == newStri:
                return True
            start += 1
        return False

File: 0567-permutation-in-string/0567-permutation-in-string.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        sort(s1.begin(), s1.end());
        if(s1.size() > s2.size()) return false;
        string newS(s2.begin(), s2.begin() + s1.size());
        string str(newS.begin(), newS.end());
        sort(str.begin(), str.end());
        if (s1 == str)
            return true;
        cout<<newS<<" ";
        for (int i = s1.size(); i < s2.size(); i++) {
            newS.erase(0, 1);
            newS += s2[i];
            string str(newS.begin(), newS.end());
            sort(str.begin(), str.end());
            if (s1 == str)
                return true;
        }
        return false;
    }
};